### DISCUSSION TITLE
How to write a file to a mounted directory?

### DISCUSSION THREAD

**Discussion by redruin1** [MCJack123/craftos2#194] [Q&A] [answered]
Hi, this is a nice piece of kit you got here. Makes development outside of Minecraft far easier and simpler.

I'm trying to have a program write a file to my hard drive through ComputerCraft's `fs` API. However, I think I'm unable to write the file because when I mount a directory to the shell it defaults to read-only (as indicated by viewing mount with no arguments).

The [documentation](https://www.craftos-pc.cc/docs/mounter) states that you can specify an optional third argument when mounting a real folder to the shell to specify whether its read-only or not, but the format of that third argument is not stated. I've tried passing 'false' as the third argument for 'readOnly', but no luck. What am I missing?

**Comment by MCJack123** [comment, marked as answer]
CraftOS-PC has a `mount_mode` configuration option to control whether you can write to mounts. By default, it's set to `ro strict`, which means writing is entirely disabled, even if the `readOnly` parameter is `false`. This is done for security reasons (mainly to avoid arbitrary scripts from modifying files outside the sandbox). You can run `config set mount_mode ro` to switch it to mount read-only by default, but still allowing you to enable writing (or use `rw` to make it writable by default, but this is a bit dangerous).

**Reply by redruin1** [reply]
Perfect, thank you very much. Safety isn't a concern at the moment since I'm only planning on running code that I've written myself. I suppose I should have scoured the configuration settings a little more astutely. Sorry for bothering you.
